test/lib/dates.lib.test.js

```diff
@@ -138,4 +138,18 @@ describe('Dates lib', () => {
       expect(result).to.be.false()
     })
   })
+
+  describe('earliestDate', () => {
+    it('should return the earliest date of 2025-04-01, 2025-03-31, 2025-03-30', () => {
+      const result = DateLib.earliestDate(['2025-04-01', '2025-03-31', '2025-03-30'])
+
+      expect(result).to.equal(new Date('2025-03-30'))
+    })
+
+    it('should be able to handle null valuse and still return the earliest date', () => {
+      const result = DateLib.earliestDate(['2025-04-01', null, '2025-03-31', '2025-03-30'])
+
+      expect(result).to.equal(new Date('2025-03-30'))
+    })
```

Comment:
I think the function as it is, suggests some additional tests we could cover in order to fully describe the behaviour we expect.

```suggestion
    let dates

    describe('given an array of date strings', () => {
      beforeEach(() => {
        dates = ['2025-04-01', '2025-03-30', '2025-03-31']
      })

      it('returns the earliest as a Date value', () => {
        const result = DateLib.earliestDate(dates)

        expect(result).to.equal(new Date('2025-03-30'))
      })
    })

    describe('given an array of date strings that contains null and undefined values', () => {
      beforeEach(() => {
        dates = ['2025-04-01', null, '2025-03-30', undefined]
      })

      it('still returns the earliest as a Date value', () => {
        const result = DateLib.earliestDate(dates)

        expect(result).to.equal(new Date('2025-03-30'))
      })
    })

    describe('given an array of date strings that only contains null and undefined values', () => {
      beforeEach(() => {
        dates = [null, undefined]
      })

      it('throws an error', () => {
        expect(() => {
          return DateLib.earliestDate(dates)
        }).to.throw('No dates provided to determine earliest')
      })
    })

    describe('given an empty array', () => {
      beforeEach(() => {
        dates = []
      })

      it('throws an error', () => {
        expect(() => {
          return DateLib.earliestDate(dates)
        }).to.throw('No dates provided to determine earliest')
      })
    })
```